python/cudf/cudf/tests/test_s3.py

```diff
@@ -34,33 +37,71 @@ def ensure_safe_environment_variables():
         os.environ.update(saved_environ)
 
 
+@pytest.fixture(scope="session")
+def s3_base(worker_id):
+    """
+    Fixture to set up moto server in separate process
+    """
+    with ensure_safe_environment_variables():
+        # Launching moto in server mode, i.e., as a separate process
+        # with an S3 endpoint on localhost
+
+        endpoint_port = (
+            "5000" if worker_id == "master" else "555" + worker_id.lstrip("gw")
+        )
```

Comment:
This approach potentially fails if pytest is run on more than 10 workers.

Reply:
Can we do integer math and then cast the integer to a string?